### ISSUE TITLE
TCX Export File Includes Imaginary HeartRateBPM Data

### ISSUE THREAD

**Issue by linalan** [GoldenCheetah/GoldenCheetah#2536]
Did not wear a HR strap. GoldenCheetah does not show any HR data for the activity. However, the exported TCX file includes a HR of 1 for every trackpoint.

                        <HeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
                            <Value>1</Value>
                        </HeartRateBpm>

GoldenCheetah 3.4
Build 3950

**Comment by Joern-R** [comment]
I just checked with the latest code. If there is no HR data series, TCX does not contain any HR data.

Please check in the Details if there is "really" no HR imported - and if there is not data series available.
Depending on the result - you can try out the latest DEV release available for download in GitHub.

**Comment by amtriathlon** [comment]
I think this was recently fixed in this commit: https://github.com/GoldenCheetah/GoldenCheetah/commit/8fed116e4d4213d35983660825a0ecb646fa8169

**Comment by linalan** [comment]
There is no HR data displayed in the Summary or Ride. HR and HrPw both show 'No data available'. Furthermore, the ride editor does not include a HR column. 

I checked Garmin Connect as well. There is no HR data displayed there. TCX file exported from Garmin Connect does not include any HeartRateBpm data.

Thus, I'm confident it is a GoldenCheetah issue. 

Latest dev build looks to have been released prior to 8fed116. Even so, I gave it a try. Appears to be a few new HR fields in the TCX, but the problem still exists.

```
                <AverageHeartRateBpm>
                    <Value>0</Value>
                </AverageHeartRateBpm>
                <MaximumHeartRateBpm>
                    <Value>0</Value>
                </MaximumHeartRateBpm>

```

```
                        <HeartRateBpm xsi:type="HeartRateInBeatsPerMinute_t">
                            <Value>1</Value>
                        </HeartRateBpm>

```

I will wait for the next dev build and test again.

**Comment by linalan** [comment]
Rode w/no HRM. Exported TCX file. Uploaded to Strava. No HR data as it should be. 

This appears to be fixed in the latest dev build (v3.5-DEV1708).
